src/System.Management.Automation/security/SecurityManager.cs

```diff
@@ -516,15 +517,27 @@ private static void UntrustPublisher(Signature signature)
             }
         }
 
+        // Check the signature via the SIP which should never erroneously validate an invalid signature
+        // or altered script.
         private static Signature GetSignatureWithEncodingRetry(string path, ExternalScriptInfo script)
         {
-            string verificationContents = System.Text.Encoding.Unicode.GetString(script.OriginalEncoding.GetPreamble()) + script.ScriptContents;
-            Signature signature = SignatureHelper.GetSignature(path, verificationContents);
+            // Invoke the SIP directly with the most simple method
```

Comment:
This comment is confusing to me, what is the simple method?  I think it should say something like 'first try to get a catalog signature, by passing in null content'.

Reply:
My understanding is that the missing catalog look up is only part of the problem, and that we should be letting the SIP try to validate with out our "help". @TravisEz13 can perhaps address additional points.

Reply:
I think getting the catalog is only half of it, like James is saying.  If we fail to get a catalog signature, we try to get the embedded signature without modifying the content.  This should be what we try first.  It is the only thing we do in Get-AuthenticodeSignature.